### ISSUE TITLE
How to report the results in your paper

### ISSUE THREAD

**Issue by youngornever** [jasonwu0731/GLMP#4]
When I run your code, sometimes I get better results (e.g. BLEU SCORE:15.45;  F1 SCORE:   0.6038309388456686; though not all F1 scores are better) than the reported scores in your paper.
Do you run several times and then average those results, or just select the best test results according to your best valid scores, as I find you do not fix a random seed in your code.
Thanks for your code release!

**Comment by jasonwu0731** [comment]
Hi @youngornever 

Thanks for your interests in our work! 

Actually, we did not try much hyper-parameters searching, so it is good that you can get a bit better results than the numbers reported in the paper! The numbers listed in the paper are averaged over two runs. If you want to have more insights, you can try to run the same hyper-parameters several times and check the variance.

**Comment by youngornever** [comment]
thanks for your quick reply!

**Comment by YaNjIeE** [comment]
@youngornever Hi, sorry to bother you. I am also reproducing the results in this paper. But with the default hyper-parameter (3-hops reasoning) that the author provided, I can only obtain 56-58.5 entity F1 scores. I am also trying other hyper-parameter settings now. Would you please tell me the hyper-parameter settings that achieve better results (e.g. BLEU SCORE:15.45; F1 SCORE: 0.6038) ?

Thanks a lot !

**Comment by youngornever** [comment]
@yanjieG  It's a long time and I dont remember the details. I only find the run.sh in my Fork. Hope it helps you.
`
echo 'set the env; need conda_env=dialog setted outside ...'
export CUDA_VISIBLE_DEVICES=1

#python3 myTrain.py -lr=0.001 -l=1 -hdd=128 -dr=0.2 -dec=GLMP -bsz=8 -ds=babi -t=1 
python3 myTrain.py -lr=0.001 -l=3 -hdd=128 -dr=0.2 -dec=GLMP -bsz=8 -ds=kvr
`

**Comment by youngornever** [comment]
@yanjieG 
I guess the env is very important. We use TITAN Xp. and torch version is 0.4.1.
Luckly, I find the checkpoint. and the test results are list as follows.

(base) user@user-ESC4000-G3-Series-Invalid-entry-length-16-Fixed-up-to-11:~/chenxiuyi/search/dialog/GLMP$ cat test.sh 
savepath=save/GLMP-KVR/HDD128BSZ8DR0.2L1lr0.001BLEU-12.15
savepath=save/GLMP-KVR/HDD128BSZ8DR0.2L1lr0.001BLEU-13.85
savepath=save/GLMP-KVR/HDD128BSZ8DR0.2L3lr0.001BLEU-14.23
python myTest.py -ds=kvr -path=${savepath} -rec=1
############### results ####################
{'dataset': 'kvr', 'task': '', 'decoder': None, 'hidden': None, 'batch': None, 'learn': None, 'drop': None, 'unk_mask': 1, 'layer': None, 'limit': -10000, 'path': 'save/GLMP-KVR/HDD128BSZ8DR0.2L3lr0.001BLEU-14.23', 'clip': 10, 'teacher_forcing_ratio': 0.5, 'sample': None, 'evalp': 1, 'addName': '', 'genSample': 0, 'earlyStop': 'BLEU', 'ablationG': 0, 'ablationH': 0, 'record': 1}
USE_CUDA: True
Reading lines from data/KVR/train.txt
Reading lines from data/KVR/dev.txt
Reading lines from data/KVR/test.txt
Read 6290 sentence pairs train
Read 777 sentence pairs dev
Read 807 sentence pairs test
Vocab_size: 1651 
Max. length of system response: 95 
USE_CUDA=True
MODEL save/GLMP-KVR/HDD128BSZ8DR0.2L3lr0.001BLEU-14.23 LOADED
STARTING EVALUATION
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 101/101 [00:18<00:00,  5.59it/s]
ACC SCORE:	0.1127633209417596
F1 SCORE:	0.6038309388456686
	CAL F1:	0.7462509407272678
	WET F1:	0.6188894233478897
	NAV F1:	0.5132461435278342
BLEU SCORE:	15.45
############### env ####################
(base) user@user-ESC4000-G3-Series-Invalid-entry-length-16-Fixed-up-to-11:~/chenxiuyi/search/dialog/GLMP$ pip list |grep torch
torch                              0.4.1      
torchvision                        0.2.1

**Comment by youngornever** [comment]
@yanjieG 
I guess the env is very important. We use TITAN Xp. and torch version is 0.4.1.
Luckly, I find the checkpoint. and the test results are list as follows.

$ cat test.sh
savepath=save/GLMP-KVR/HDD128BSZ8DR0.2L1lr0.001BLEU-12.15
savepath=save/GLMP-KVR/HDD128BSZ8DR0.2L1lr0.001BLEU-13.85
savepath=save/GLMP-KVR/HDD128BSZ8DR0.2L3lr0.001BLEU-14.23
python myTest.py -ds=kvr -path=${savepath} -rec=1
############### results ####################
{'dataset': 'kvr', 'task': '', 'decoder': None, 'hidden': None, 'batch': None, 'learn': None, 'drop': None, 'unk_mask': 1, 'layer': None, 'limit': -10000, 'path': 'save/GLMP-KVR/HDD128BSZ8DR0.2L3lr0.001BLEU-14.23', 'clip': 10, 'teacher_forcing_ratio': 0.5, 'sample': None, 'evalp': 1, 'addName': '', 'genSample': 0, 'earlyStop': 'BLEU', 'ablationG': 0, 'ablationH': 0, 'record': 1}
USE_CUDA: True
Reading lines from data/KVR/train.txt
Reading lines from data/KVR/dev.txt
Reading lines from data/KVR/test.txt
Read 6290 sentence pairs train
Read 777 sentence pairs dev
Read 807 sentence pairs test
Vocab_size: 1651
Max. length of system response: 95
USE_CUDA=True
MODEL save/GLMP-KVR/HDD128BSZ8DR0.2L3lr0.001BLEU-14.23 LOADED
STARTING EVALUATION
100%|█████████████████████████████████████| 101/101 [00:18<00:00, 5.59it/s]
ACC SCORE: 0.1127633209417596
F1 SCORE: 0.6038309388456686
CAL F1: 0.7462509407272678
WET F1: 0.6188894233478897
NAV F1: 0.5132461435278342
BLEU SCORE: 15.45
############### env ####################
$ pip list |grep torch
torch 0.4.1
torchvision 0.2.1

**Comment by YaNjIeE** [comment]
@youngornever 

Thank you for your quick response and the information your provide. I will try these hyper-parameters above. That help a lot.

 Have a good day :)
